kratos/utilities/python_function_callback_utility.h

```diff
@@ -8,214 +8,357 @@
 //                   Kratos default license: kratos/license.txt
 //
 //  Main authors:    Riccardo Rossi
+//  Collaborator:    Vicente Mataix Ferrandiz
 //
 
 #if !defined(KRATOS_PYTHON_FUNCTION_CALLBACK_UTILITY_H_INCLUDED)
 #define  KRATOS_PYTHON_FUNCTION_CALLBACK_UTILITY_H_INCLUDED
 
+// System includes
+#include <cmath>
+
+// External includes
 #include <pybind11/pybind11.h>
 #include <pybind11/eval.h>
 
-#include <cmath>
+// Project includes
 #include "includes/define.h"
 #include "includes/kratos_parameters.h"
 #include "includes/model_part.h"
 
 namespace Kratos
 {
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
 
+///@}
+///@name  Functions
+///@{
 
-/**this function allows to call a function method of type f(x, y, z, t) implemented in python.
-* NOTE: this makes this file to depend on python
-*
-* the functions can be constructed by providing a python-defined method of the type
-*
-*  class aux_object_cpp_callback:
-*    def __init__(self, function_string ):
-*        #TODO: check python version
-*        self.compiled_function = compile(function_string, '', 'eval', optimize=2)
-*
-*    def f(self,x,y,z,t):
-*        return  eval(self.compiled_function)
-*
-* the object is then insantiated as
-* aux_function = PythonGenericFunctionUtility(aux_object_cpp_callback(self.function_string))
-*
-* optionally one can specify a rotation matrix and an origin so that the function can be defined in a rotated system of coordinates
-*/
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class PythonGenericFunctionUtility
+ * @ingroup KratosCore
+ * @brief This function allows to call a function method of type f(x, y, z, t) implemented in python.
+ * @details The functions can be constructed by providing a python-defined method of the type
+ *
+ *  class aux_object_cpp_callback:
+ *    def __init__(self, function_string ):
+ *        #TODO: check python version
+ *        self.compiled_function = compile(function_string, '', 'eval', optimize=2)
+ *
+ *    def f(self,x,y,z,t):
```

Comment:
in your proposal this becomes a larfer function, right?

Reply:
Yes, but this is transition, I am thinking to do a future PR allowing any type of variable. I do not do that here because I assume it will require further discussion